packages/admin/cms-admin/src/dam/DataGrid/fileUpload/useFileUpload.tsx

```diff
@@ -47,8 +47,18 @@ interface Files {
     fileRejections: FileRejection[];
 }
 
+interface ImportedSource {
+    sourceId: string;
+    sourceType: string;
+}
+
+interface UploadFileOptions {
+    folderId?: string;
+    importedFile?: ImportedSource;
```

Comment:
This should be called `importSource`, and the type be `type ImportSource = { id: string; type: string }`

Or you could pass `importSourceType` and `importSourceId` to be in line with the API (have I suggested changing it to an object before? 😅 ).

The type of the options could be changed to achieve some sort of validation:

```ts
type UploadFileOptions =
  { folderId?: string; } 
  & (
       { importSourceType?: never; importSourceId?: never; } |
       { importSourceType: string; importSourceId: string; }
    );
```

(haven't tried this)